packages/@azure/servicebus/data-plane/test/streamingReceiver.spec.ts

```diff
@@ -842,3 +842,118 @@ describe("Streaming - User Error", function(): void {
     await testUserError();
   });
 });
+
+describe("Streaming - maxConcurrentCalls", function(): void {
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function testConcurrency(maxConcurrentCalls?: number): Promise<void> {
+    if (
+      typeof maxConcurrentCalls === "number" &&
+      (maxConcurrentCalls < 1 || maxConcurrentCalls > 2)
```

Comment:
Values 1 and 2 are looking too specific. Would be more ideal to have a test that covers expected behavior where it takes an arbitrary parameter number.

Reply:
Like a random number in 10-50 each time ?

Reply:
What I intended to cover here is the case of no concurrency (value : 1) and a case of concurrency (value:2) and the default case when no value is provided.

I didn't want to add a test for higher numbers because the set up would get more complicated. To test say for a number `n`, we should send `n+1` messages, and check that the (n+1)th message is not processed before atleast 1 of the messages are settled.

As a middle ground, I could remove the `if` condition above and replace the `maxConcurrentCalls === 2` check a few lines below with `maxConcurrentCalls > 1` and the tests would still pass. And update all cases that use `2` to use a number greater than 1

Reply:
Does that come under 'stress and perf testing' ?

Reply:
Removed the check for values 1 or 2 in https://github.com/Azure/azure-sdk-for-js/pull/1317/commits/7399c74e49deb5be326f6cbfc71fa5be62545d3c